### ISSUE TITLE
Web Interface for search results and filter options

### ISSUE THREAD

**Issue by blackforestboi** [WorldBrain/Research-Engine#24]
Currently users can only retrieve 5 results at a time via the drop down of the address bar and have limited filter options. Right now I can only filter for the time and this also only by typing i.e. after:"yesterday".

We need a relatively simple web interface that can be run locally and lets users filter for various metrics:
- Keywords (via search bar)
- Timeframe
- Author
- Source

Mockup:
![mockup_results](https://cloud.githubusercontent.com/assets/7870039/20463920/c6d97ccc-af3d-11e6-84ff-29b0323f9c18.png)

**Comment by blackforestboi** [comment]
note: 

Before starting, we should make sure, that we prepare this web view is later also able to display other sources (like evernote results etc)

We don't have to do it now, as we want to develop whats ultimately necessary for now, but keep in mind, that we will add more sources later, that are not web results.

**Comment by blackforestboi** [comment]
@obsidianart @hoshomoh 
What is your stance on implementing this part (and refactor the other assets) in React. 

This way we can easily port it to mobile and other platforms later.

**Comment by obsidianart** [comment]
I would personally use react, but I don't have a strong reason for it instead of something else (at least for now).

**Comment by blackforestboi** [comment]
It's also important to consider the future requirements of the tool:

1. We will have mobile applications for iOS and Android 
2. We will have a native OS application (OSX, Win, Linux), possibly as an Electron App
3. We will have parts of the software run server side. 

Any other requirements that come to mind?

So a framework will be useful that allows us to not redo work for each platform we develop our tool. 
React seems suitable. Anybody sees downsides to it or limitations?

**Comment by blackforestboi** [comment]
@amitkumarj441 @treora @hoshomoh @obsidianart

**Comment by blackforestboi** [comment]
Ok, So @obsidianart and me have just been spending our evening holding hands and setting up the base react framework.... :)

@aquibm Now we can start working on this part. 

Important: check the #82 and give some feedback on the chosen architecture before starting. Thanks!

Basically what we need is a simple results view that covers the current functionalities of the address bar search.
It does not need to be connected to the actual functionalities yet. 

It just needs to be able to input:
 - the search query
 - time picker with two fields

and output:
 -  Title of page
 -  URL of page
 -  Text of page with highlights (highlights are coming from pouchDB-Quick Search)

#### Updated Mockup:

![mockup_resultspage](https://cloud.githubusercontent.com/assets/7870039/22186599/5a23814e-e0f8-11e6-9785-00b5833c88a2.png)

**Comment by obsidianart** [comment]
I would try to keep the design the same (sidebar and header) and add this functionality within the routed view. For getting data from pouchdb you can use a middleware (which is the most elegant solution) but I would keep it simple and call the functionality from what we have (within the action/reducer). #82 has a nice approach to reducer keeping them at component level, I've never used it but I can appreciate the idea behind it.

**Comment by blackforestboi** [comment]
> I would try to keep the design the same (sidebar and header)

Good idea! 

Question here: can we call the results page then via a short cut/a fixed URL? This would greatly enhance the work-flow for people. 
Also necessary because one option to enter the view is to chose the first result from the drop down, which then opens the web view. 

As it seemed with `memoryHistory`, we dont have specific URLs for each the views anymore?

**Comment by obsidianart** [comment]
I think we can go back to hashHistory, the problem memoryHistory was supposed to solve didn't change. memoryHistory is probably good for the popup.html (which we don't need).

**Comment by blackforestboi** [comment]
Oh I thought memoryHistory did solve the problem?

But whatever is clever then. the search page just needs to be able to be callable with a URL then. Can we also pass the query via this url?

**Comment by obsidianart** [comment]
nope, we solved the problem in another way. you can pass the query in the forma #/search/query

**Comment by ShamariFeaster** [comment]
I'm guessing react is the current choice because it's isomorphic? My thoughts would be that if what the web interface has to do is relatively simple (display results in as table, paginate, etc) then maybe react is overkill? I'd be curious to see, concretely, just how isomorphic a react code base really is. Maybe something like Vue.js might be a better fit?

**Comment by obsidianart** [comment]
Being isomorphic (or better, universal) is normally used for a javascript code which can be used server side or client side, there's no server here so I would say it's not one of the reason.
React is quite simple and we have no download size issues(it stays in memory), redux might be overkilled for the added complexity.
I think the main reason to choose React is that we have people who know React :), and I've never used vue.js.
Looking at Vue.js I can't see any routing, so probably it's an addition to another templating system we don't have, like php (I don't want to create multiple html file with the same code).
I've read some comparison between Vue and React and I feel they compare only the data binding layer, not the ecosystem. We can probably use redux+routing+vue to achieve the same result (I saw angular+redux), but I would need to research more.
Ultimately, if between 2 choices one is known by more people, chose the latter. It's easier to find people who understand React and once it's set up you need to know very little.

**Comment by blackforestboi** [comment]
Hey @ShamariFeaster 

thanks for the input. 
I think one other reason for React is that we easily can port it then to other devices. 
In the near future, we want to build an Electron, iOS and Android app. For that we then don't have to completely redo the whole code.

**Comment by blackforestboi** [comment]
@raj-maurya 

Do you want to take on this task?

**Comment by fherrador** [comment]
So, essentially, is there a limit actually to 5 results coming from the URL bar-combo?
Do this means that the only way to overcome this is to build a local result page to show more?
Don't we have something at hand to overcome this while we wait to the new implementation?

**Comment by blackforestboi** [comment]
Hey @fherrador 

Thanks for letting us know about your issue with that...we have it too :(

Unfortunately we cannot modify the amount of results shown in the omnibox([Source](http://stackoverflow.com/questions/8694725/increasing-the-number-of-suggests-shown-in-omnibox))
But the update will be available pretty soon, it's the next thing we work on after the refactoring of the current code. 

 I am sorry to not give you a more encouraging answer :/
